d3a_interface/kafka_communication/kafka_producer.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import json
+from zlib import compress
+from kafka import KafkaProducer
+
+from d3a_interface.kafka_communication import KAFKA_URL, DEFAULT_KAFKA_URL, KAFKA_USERNAME, \
+    KAFKA_PASSWORD, KAFKA_COMMUNICATION_SECURITY_PROTOCOL, KAFKA_SASL_AUTH_MECHANISM, \
+    KAFKA_API_VERSION, create_kafka_new_ssl_context, KAFKA_TOPIC
+
+KAFKA_PUBLISH_RETRIES = 5
+KAFKA_BUFFER_MEMORY = 2048000000
+KAFKA_MAX_REQUEST_SIZE = 2048000000
+
+
+def kafka_connection_factory():
+    try:
+        return KafkaConnection()
+    except Exception:
+        logging.info("Running without Kafka connection for simulation results.")
```

Comment:
When is this supposed to happen ?

Reply:
Typically in 2 circumstances:
- when running d3a using the CLI 
- when running integration tests 
During these 2 usecases, we do not need Kafka at all. Therefore the factory method is responsible for constructing the DisabledKafkaConnection class.

Reply:
Understood, thanks!